Fix: Light theme on profile screen and ui enhancement.

hemang-mishra opened the pull request:
These are the changes made in this PR.
1. Fixed the compatibility box to occupy the full width of screen.
2. Changed the text color of "You have no common artists" to onSurface for proper visibility in both dark and light themes.
3. Made the corners of addListenButton, musicBrainButton, Follow and unfollow button rounded with 4.dp.
4. Also added a padding of 4.dp below this row, to enhance the ui.

Do let me know if any changes have to be made.

### Screenshots:
Issue screenshot:
![compat_light_unfixed](https://github.com/user-attachments/assets/31048565-14f7-4810-a539-55f13546d514)

Fixed light screen:
![compat_light_fixed](https://github.com/user-attachments/assets/0a42272c-dc62-4f88-b7e3-1549f527b202)

Fixed dark mode screen:
![compat_dark_fixed](https://github.com/user-attachments/assets/c83a2de1-93a4-4e5e-823e-63f147b9cde3)


Linked to issue: [Mobile-205](https://tickets.metabrainz.org/projects/MOBILE/issues/MOBILE-205?filter=allopenissues)

07jasjeet commented:
Hi @hemang-mishra, Thanks for opening this PR! As a side note, you can chat the whole team by going through this guide https://wiki.musicbrainz.org/Communication and setting up your favorable mode of communication.

07jasjeet commented:
Adding to this issue, can you try to use [this](https://github.com/metabrainz/listenbrainz-android/blob/main/app/src/main/java/org/listenbrainz/android/ui/components/FollowButton.kt) component. We can implement this incrementally in future PRs or this one itself, according to your liking :) This would be a great problem to solve!

hemang-mishra commented:
Thanks @07jasjeet , I have joined the discord server, I will definitely reach out to the community on the development channel. I would love to work on this component as a separate PR. I tried implementing it here but seems the state of the button is not changing on click. I will start working on this asap.
Just to confirm, we need to replace the old follow and unfollow button on the profile page with this component, right?

hemang-mishra commented:
I have implemented mentioned component.
I have fixed the state issues in the view model.

Please let me know if any furthur changes are required.

The screens look like this:
![follow1](https://github.com/user-attachments/assets/ab1b900e-de5f-4460-a908-0b27f21c5eee)
![follow2](https://github.com/user-attachments/assets/31c180b7-994e-4ffd-bfbd-5b59a225aec6)

hemang-mishra commented:
If required, I can change the color of this button to match color of MusicBrainz button.
For that I have to modify the Follow button component.
Should I do that?

07jasjeet commented:
@hemang-mishra Yes, please pass an argument `buttonColor = lb_purple`

07jasjeet commented on app/src/main/java/org/listenbrainz/android/ui/screens/profile/BaseProfileScreen.kt:
```diff
@@ -187,14 +187,20 @@ fun BaseProfileScreen(
-                                })
-                            }
+                            false->
+                                Box() {
```
Please remove unnecessary `Box` wrapper.

07jasjeet replied:
I see, this is a fault in the component itself, have a look if you can fix it ;)

07jasjeet commented on app/src/main/java/org/listenbrainz/android/ui/screens/profile/BaseProfileScreen.kt:
```diff
@@ -187,14 +187,20 @@ fun BaseProfileScreen(
-                            }
+                            false->
+                                Box() {
+                                    org.listenbrainz.android.ui.components.FollowButton(
```
Please import the component itself.

hemang-mishra commented:
@07jasjeet Done updated color.

07jasjeet commented on app/src/main/java/org/listenbrainz/android/viewmodel/UserViewModel.kt:
```diff
@@ -71,13 +71,17 @@ class UserViewModel @Inject constructor(
         if(username.isNullOrEmpty()) return
         viewModelScope.launch (ioDispatcher) {
             socialRepository.followUser(username)
```
Can we change the state instantly and revert back if something goes wrong? Here:
```kotlin
listenStateFlow.value = listenStateFlow.value.copy(isFollowing = true)
val result = socialRepository.followUser(username)
if (result.status == Resource.Status.FAILED) {
    listenStateFlow.value = listenStateFlow.value.copy(isFollowing = false)
}
```

This way, we would have quick response to user tap, but in reality, we would still be processing api request in the background. If the request fails, we simply reflect that to the UI at a later stage.

07jasjeet commented on app/src/main/java/org/listenbrainz/android/viewmodel/UserViewModel.kt:
```diff
@@ -71,13 +71,17 @@ class UserViewModel @Inject constructor(
     fun unfollowUser(username: String?){
         if(username.isNullOrEmpty()) return
         viewModelScope.launch(ioDispatcher) {
             socialRepository.unfollowUser(username)
```
Same here as above!

07jasjeet commented on app/src/main/java/org/listenbrainz/android/ui/screens/profile/BaseProfileScreen.kt:
```diff
@@ -192,6 +192,7 @@ fun BaseProfileScreen(
                                     org.listenbrainz.android.ui.components.FollowButton(
                                         modifier = Modifier,
                                         isFollowedState = uiState.listensTabUiState.isFollowing,
+                                        buttonColor = Color(0xFF353070),
```
Please import `lb_purple` from `Color.kt` :)

hemang-mishra commented:
@07jasjeet I have made changes as you requested.
Removing box is giving error in UI. 

Also the text color is still black, should I change the original FollowButton composable
![image](https://github.com/user-attachments/assets/801bc9b8-eaeb-485d-b105-2c2a78dc3b1a)

hemang-mishra commented:
I have fixed the text color issue by slightly modifying the component. Please let me know if any other changes are required.

Here is an image of the button.
![image](https://github.com/user-attachments/assets/796ba1ab-d51d-43e6-bde1-9e2a33091715)

07jasjeet commented:
Looks great @hemang-mishra! LGTM
